src/validation.h

```diff
@@ -798,6 +797,20 @@ class CChainState
 
     bool LoadBlockIndexDB() EXCLUSIVE_LOCKS_REQUIRED(cs_main);
 
+    //! Indirection necessary to make lock annotations work with an optional mempool.
+    RecursiveMutex* MempoolMutex() const LOCK_RETURNED(m_mempool->cs)
+    {
+        return m_mempool ? &m_mempool->cs : nullptr;
+    }
+
+    void UpdateMempoolForReorg(
+        DisconnectedBlockTransactions& disconnectpool,
+        bool fAddToMempool) EXCLUSIVE_LOCKS_REQUIRED(cs_main, m_mempool->cs);
+
+    /** Check warning conditions and do some notifications on new chain tip set. */
+    void UpdateTip(const CBlockIndex* pindexNew, CChainState& active_chainstate)
```

Comment:
No need for the `CChainState& active_chainstate` argument - this is a private method in `CChainState` where the `active_chainstate` arg is always `*this`. You can remove the argument and use the implicit `this` inside the function.

Reply:
Good call - weird, thought I had made that change... maybe lost in rebase shuffle.